### DISCUSSION TITLE
New 3.1.1 out now

### DISCUSSION THREAD

**Discussion by petretiandrea** [petretiandrea/home-assistant-tapo-p100#739] [Announcements]
I'm really happy to announce that version 3.1.1 is out now, with a lot of stability improvements, better support to tapo devices with components design! 🥳 Also add diagnostic report, to help me to help you, and new firmware management offically supported by HA.

Thanks to testing, plupg100 library is more stable, of course, there can always be problems to fix, since it is reverse engineering.

I hope in few next days to make this integration the official one on HA. There is still much work to be done!

Thanks everyone for support 🙏 

EDIT: please refer to 3.1.1 with a minor fix 🙏

**Comment by tjorim** [comment]
Is it not compatible with the HA core `tplink` integration? I was using them side-by-side to compare without any issues (on dev1). Since updating to the stable 3.1.0 release of this custom integration, I keep getting repairs for the core one. Any idea why? `Authentication failed while fetching 192.168.0.x data: Got a security error from 192.168.0.x after handshake completed`

**Reply by petretiandrea** [reply]
No, there is no interference between two integration. Maybe authenticating one device to one integration, can lead to unauthorize to other, but as I say, maybe
